specs/Standards/Tokens/FungibleTokenMetadata.md

```diff
@@ -1,60 +1,45 @@
 # Fungible Token Metadata ([NEP-148](https://github.com/near/NEPs/discussions/148))
 
-Version `1.0.0`
-
-## Summary
-[summary]: #summary
-
-An interface for a fungible token's metadata. The goal is to keep the metadata future-proof as well as lightweight. This will be important to dApps needing additional information about an FT's properties, and broadly compatible with other tokens standards such that the [NEAR Rainbow Bridge](https://near.org/blog/eth-near-rainbow-bridge/) can move tokens between chains.
-
-## Motivation
-
-Custom fungible tokens play a major role in decentralized applications today. FTs can contain custom properties to differentiate themselves from other tokens or contracts in the ecosystem. In NEAR, many common properties can be stored right on-chain. Other properties are best stored off-chain or in a decentralized storage platform, in order to save on storage costs and allow rapid community experimentation.
-
-As blockchain technology advances, it becomes increasingly important to provide backwards compatibility and a concept of a spec. This standard encompasses all of these concerns.
+## Reference-level explanation
 
-Prior art:
+A fungible token contract implementing the metadata standard shall contain a field named `ft_metadata`. The field will link to a structure with the properties from the interface below.
 
-- [EIP-1046](https://eips.ethereum.org/EIPS/eip-1046)
-- [OpenZeppelin's ERC-721 Metadata standard](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Metadata) also helped, although it's for non-fungible tokens.
+**Interface**:
 
-## Guide-level explanation
+```ts
+type FungibleTokenMetadata = {
+    spec: string;
+    name: string;
+    symbol: string;
+    defaultIcon: bytes|null;
+    darkIcon: bytes|null;
+    reference: string|null;
+    reference_hash: string|null;
+    decimals: number;
+}
+```
 
-A fungible token smart contract allows for discoverable properties. Some properties can be determined by other contracts on-chain, or return in view method calls. Others can only be determined by an oracle system to be used on-chain, or by a frontend with the ability to access a linked reference file.
+# Discussion
 
-### Examples scenario
+## On-chain vs. offchain hosted data
 
-#### Token provides metadata upon deploy and initialization
+Currently deployment on a NEAR token smart contract is 200 kb. The cost of 5 kb metadata on the cost of code deployment on the top of the contract deployment is miniscule. Icons are in the range of 5 kBytes - 200 kBytes per icon, so they are comparable to a smart contract deployment. 
 
-Alice deploys a wBTC fungible token contract.
+Off-chain data comes complicated with privacy and tracking issues as pointed out by Robert Zarembe.
 
-**Assumptions**
+NEAR is the best blockchain in the world. For the reducing complexity, and after discussing with people on Discord, my stance is that everything on-chain. NEAR should scale for this as it scales for many things.
```

Comment:
I had a bit different view:
+ there are other technologies which may be more suitable for static content: Sia, Filecoin, IPFS.... I don't think NEAR wants to be a storage layer. It's not efficient to scale all use-cases with same technology.
+ So, putting _everything_ on-chain is too far reaching.  Even icon - rember there will be many icons. Not only black-white. But some where differ with size or composition (with text, without text ...).